zulip_bots/zulip_bots/test_lib.py

```diff
@@ -82,31 +83,19 @@ def check_expected_responses(self, expectations, expected_method='send_reply',
             trigger_messages.append(stream_message_dict)
 
         for trigger_message in trigger_messages:
-            if storage is None:
-                current_storage = None
-            else:
-                # A new (copy of) the StateHandler is used for each trigger message.
-                # This avoids type="all" failing if state is created in the first iteration.
-                current_storage = deepcopy(storage)
-
```

Comment:
I don't like re-introducing this landmine.  I think instead we should just change the mechanism for `all` to run through the whole block below twice (clearing the storage in between), one for private messages, the other for stream, rather than having it potentially alternate.

Reply:
Okay. I think we should strongly encourage test writers to use this function only once per test case when testing stateful bots: Even if we clear the state between runs through private and stream messages, the exit state is not clear to the user: It could be either the state after sending the stream messages, or the state after the private messages, if he doesn't explicitly look into the function's interior.